core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -680,12 +680,15 @@ abstract class AbstractFetcherThread(name: String,
        */
       val offsetAndEpoch = leader.fetchEarliestOffset(topicPartition, currentLeaderEpoch)
       val leaderStartOffset = offsetAndEpoch.offset
-      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to current " +
-        s"leader's start offset $leaderStartOffset")
       val offsetToFetch = Math.max(leaderStartOffset, replicaEndOffset)
+      info(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to " +
+        s"new fetch offset $offsetToFetch")
       // Only truncate log when current leader's log start offset is greater than follower's log end offset.
-      if (leaderStartOffset > replicaEndOffset)
+      if (leaderStartOffset > replicaEndOffset) {
         truncateFullyAndStartAt(topicPartition, leaderStartOffset)
+        warn(s"Truncate fully and start at leader's start offset $leaderStartOffset for partition $topicPartition " +
+          s"because of local replica logEndOffset $replicaEndOffset smaller than leader's start offset.")
```

Comment:
nit: maybe "because of local replica" => "because the local replica"?

Reply:
nice catch! thanks!